File: java/src/org/gessinger/gepard/xmp/FileSender.java
package org.gessinger.gepard.xmp ;
import org.gessinger.gepard.* ;

import java.util.List ;

public class FileSender
{
  static public void main ( String[] args )
  {
    Util.argsToProperties ( args ) ;
    try
    {
      FileSender j = new FileSender() ;
    }
    catch ( Exception exc )
    {
      exc.printStackTrace() ;
    }
  }
  FileSender()
  throws Exception
  {
    final Client client = Client.getInstance() ;

    Event event = new Event ( "__FILE__" ) ;
    final String file = Util.getProperty ( "file", "build.xml" ) ;
    event.putValue ( "DATA", new FileContainer ( file ) ) ;
    client.request ( event, new ResultCallback()
    {
      public void result ( Event e )
      {
        if ( e.isBad() )
        {
          System.out.println ( e ) ;
        }
        else
        {
          System.out.println ( "File " + file + " sent successfully." ) ;
          System.out.println ( "code: " + e.getStatusCode() );
          System.out.println ( "name: " + e.getStatusName() );
          System.out.println ( "reason: " + e.getStatusReason() );
        }
        client.close() ;
      }
    }) ;
  }
}

File: python/test/FileSender.py
#!/usr/bin/env python

import os, sys, datetime
sys.path.insert ( 0, os.path.dirname(os.path.abspath(__file__) ) + "/../" )
import gepard

client = gepard.Client.getInstance()

event = gepard.Event ( "__FILE__" )

fileName = gepard.util.getProperty ( "file", "data.bin" )
print ( "fileName=" + str ( fileName ) )
fr = gepard.FileReference ( fileName )
event.putValue ( "FR", fr )

def result ( e ):
	print ( "callback for __FILE__" )
	FR = e.getValue ( "FR" )
	print ( type(FR["data"]) )
	# print ( e )
	# if e.isBad():
	# 	print ( e.getStatusReason() )
	# else:
	# 	print ( e.getValue ( "file_list" ) )
	e.getClient().close()

# Using a name only
client.request ( event, result )
